### ISSUE TITLE
Long trip route is ...weird

### ISSUE THREAD

**Issue by Karmalakas** [organicmaps/organicmaps#9026]
**Steps to reproduce**
1. Select _Vilnius, Lithuania_ (54.69, 25.28) as a _Route from_ 
1. Select _Barcelona, Spain_ (41.38, 2.18) as a _Route to_

**Expected behaviour**
Route NOT leading through a hostile country (which also makes route longer)

**Screebshots**
OSM route
![image](https://github.com/user-attachments/assets/559b7d62-d33a-498f-9deb-e663bc088b81)

Google route
![image](https://github.com/user-attachments/assets/add69265-5a8d-48cb-ada3-011a618ed527)

OM route
![image](https://github.com/user-attachments/assets/71176748-8626-4f25-a2f8-1093d87153d6)

OM routes from EU to Russia and back to EU. Also the route is longer both in distance and time (related issue - #9025)

**System information:**
 - Operating system and its version: Android 14
 - Organic Maps version: 2024.08.17-4-Google
 - Device Model: Pixel 6a

**Comment by biodranik** [comment]
There are no border-related preferences or restrictions yet in OM. I think there was an issue about it already.

**Comment by Karmalakas** [comment]
Yes I linked the related issue. Wasn't sure if it's the same or something else

**Comment by biodranik** [comment]
No, the original issue is this one: https://github.com/organicmaps/organicmaps/issues/774

**Comment by Karmalakas** [comment]
I see. But in this case it wouldn't help, as I'd need to cross multiple borders anyway.

In any case, I believe this issue can be closed then 🤔
